### ISSUE TITLE
Feature: Nested mocking objects

### ISSUE THREAD

**Issue by ruXlab** [mockk/mockk#244]
I found that sometimes it's impossible to create mock while creating mock. 

For instance, this code below throws an exception
```
mockk<User> {
    every { name } returns "Bob"
    every { address } returns mockk {
        every { street } returns "Old Street"
        every { zip } returns "EC1"
    }
}
```

#### Stacktrace 
```
io.mockk.MockKException: no answer found for: User(#1).getAddress()

	at io.mockk.impl.stub.MockKStub.defaultAnswer(MockKStub.kt:99)
	at io.mockk.impl.stub.MockKStub.answer(MockKStub.kt:51)
	at io.mockk.impl.recording.states.AnsweringState.call(AnsweringState.kt:16)
	at io.mockk.impl.recording.states.AnsweringStillAcceptingAnswersState.call(AnsweringStillAcceptingAnswersState.kt:16)
	at io.mockk.impl.recording.CommonCallRecorder.call(CommonCallRecorder.kt:54)
	at io.mockk.impl.stub.MockKStub.handleInvocation(MockKStub.kt:272)
	at io.mockk.impl.instantiation.JvmMockFactoryHelper$mockHandler$1.invocation(JvmMockFactoryHelper.kt:25)
	at io.mockk.proxy.jvm.advice.Interceptor.call(Interceptor.kt:20)
	at io.mockk.proxy.jvm.advice.BaseAdvice.handle(BaseAdvice.kt:42)
	at io.mockk.proxy.jvm.advice.jvm.JvmMockKProxyInterceptor.interceptNoSuper(JvmMockKProxyInterceptor.java:45)
	at MockInsideMockTest$User$Subclass0.getAddress(Unknown Source)
	at MockInsideMockTest.create mock inside mocks(MockInsideMockTest.kt:28)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
```

### Expected Behavior

Allow to create nested mocks

### Current Behavior

`MockKException` is thrown if nested mocks are created. Mocks intitalized independently are working fine (see unit test below)

#### Steps to Reproduce

Run this test:
```
import io.mockk.every
import io.mockk.mockk
import org.junit.Assert.assertNotNull
import org.junit.Test

class MockInsideMockTest {

    interface User {
        val name: String
        val address: Address
    }
    interface Address {
        val street: String
        val zip: String
    }

    @Test
    fun `create mock inside mocks`() {
        val bob = mockk<User> {
            every { name } returns "Bob"
            every { address } returns mockk {
                every { street } returns "Old Street"
                every { zip } returns "EC1"
            }
        }

        assertNotNull(bob)
        assertNotNull(bob.address)
    }

    @Test
    fun `create mock mocks separately`() {
        val bobAddress = mockk<Address> {
            every { street } returns "Old Street"
            every { zip } returns "EC1"
        }
        
        val bob = mockk<User> {
            every { name } returns "Bob"
            every { address } returns bobAddress
        }

        assertNotNull(bob)
        assertNotNull(bob.address)
    }
}
````

#### Context

* MockK version: 1.0
* OS: Ubuntu 18.04
* Kotlin version: 1.3.20
* JDK version: OpenJDK Runtime Environment (build 11.0.1+13-Ubuntu-3ubuntu3.18.10.1)
* JUnit version: 4.12
* Type of test: unit test

**Comment by oleksiyp** [comment]
Yes mockk doesnt support such hierarchy. Extract variable in returns.
There were few similar reports.

Probably it is possible to implement it, but need sufficient rework

**Comment by oleksiyp** [comment]
Done in a774f4a and f0e0673

**Comment by ruXlab** [comment]
Thank you for your time @oleksiyp !
Will try next release

**Comment by oleksiyp** [comment]
Released [1.9.1](https://github.com/mockk/mockk/releases/tag/1.9.1). Please try it out and close the ticket if everything is fine. Official docs will be updated a little bit later

**Comment by ruXlab** [comment]
:tada: :tada: :tada: :tada: 

Thanks a lot, it works as expected now, very handy. 
Appreciate your time and effort!
